### ISSUE TITLE
[BUG] auditlog complaints ALL THE TIME in logs 

### ISSUE THREAD

**Issue by KlavsKlavsen** [opensearch-project/OpenSearch#1724]
**Describe the bug**
I get these error messages in log output (in opensearch 1.2.1 installed via helm chart on k8s):
```
[2021-12-14T15:07:57,149][ERROR][o.o.s.a.s.InternalOpenSearchSink] [opensearch-cluster-master-0] Unable to index audit log {"audit_cluster_name":"opensearch-cluster","audit_transport_headers":{"_system_index_access_allow 
 opensearch org.opensearch.index.IndexNotFoundException: no such index [security-auditlog-2021.12.14]
```
and
```
[2021-12-14T15:17:05,556][DEPRECATION][o.o.d.c.m.IndexNameExpressionResolver] [opensearch-cluster-master-0] this request accesses system indices: [.opendistro_security], but in a future major version, direct access to system indices will be prevented by default                                                                                                                                                                                               
```
I just installed it and ran graylog against it.. don't know if its related to what graylog does?

**Comment by dblock** [comment]
Any indication whether graylog causes this? Maybe help confirm that by turning it off/on and asking in https://community.graylog.org/ first?

**Comment by KlavsKlavsen** [comment]
its graylog that does something stupid :(

**Comment by dblock** [comment]
@KlavsKlavsen Care to add some detail for the next person that runs into this? Thanks

**Comment by KlavsKlavsen** [comment]
@dblock unfortunately I have no idea what its doing and haven't solved it. I rolled up a seperate opensearch instance - doing nothing to it - to see if that caused any of these logs and it didn't.. So its the "use" of opensearch (which here is done by graylog) - thats causing it.. )

**Comment by pierre-mallet** [comment]
Hello, for information we ran into the same problem on our clusters, 

it was due to a "action.auto_create_index: false" settings for our cluster. So the audit indices were not able to be created on the fly. 
We resolved this with by allowing the security audit indices to be auto created.

`
PUT _cluster/settings
{
  "persistent": {
    "action.auto_create_index": "security-auditlog-*" 
  }
}`

Hope it will help the next folks having the same problem. 

Regards
